pyteal/ast/itxn.py

```diff
@@ -249,6 +249,63 @@ def SetFields(cls, fields: dict[TxnField, Expr | list[Expr]]) -> Expr:
         fieldsToSet = [cls.SetField(field, value) for field, value in fields.items()]
         return Seq(fieldsToSet)
 
+    @classmethod
+    def ExecuteMethodCall(
+        cls,
+        *,
+        app_id: Expr,
+        method_signature: str,
+        args: list[abi.BaseType | Expr | dict[TxnField, Expr | list[Expr]]],
+        extra_fields: dict[TxnField, Expr | list[Expr]] = {},
```

Comment:
wait... is `{}` mutable? consider using `None` as default and internally cast to `dict`, and it would apply to `MethodCall`...?

Reply:
`MethodCall` also needs to be updated with this. I'm not sure how the existing code works without this 🤔

Reply:
^ `MethodCall` argument `extra_fields` is also default to be `{}`... my bad didn't catch it before.

Reply:
ah, I'll need one of you to explain to me why this is bad plz

I think I've changed it to what it should be now